### PR TITLE
Localization 3.0 Updates

### LINKED ISSUE #14225: 3.0 update: Globalization and localization
Ryan to do this.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d1833676-ff1b-c587-cc9c-81afffa8ef68
* Version Independent ID: ad989395-3f49-4d55-521e-cf6fe56281ee
* Content: [Globalization and localization in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/localization?view=aspnetcore-3.0#feedback)
* Content Source: [aspnetcore/fundamentals/localization.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/localization.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by hishamco**
Fix #14225
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/localization?view=aspnetcore-3.0&branch=pr-en-us-14874)

- [x] Add note for unsupported log level change
- [x] Support `Content-Language` header

/cc @ryanbrandenburg @Rick-Anderson

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
Either move `'~/includes/currency.md'` back or or fix the other two files.


Severity | Type | Code (click link for details) | File | Message | Line | Time
-- | -- | -- | -- | -- | -- | --
Error | User | include-not-found | aspnetcore/tutorials/first-mvc-app/validation.md | Cannot resolve '~/includes/currency.md' relative to 'tutorials/first-mvc-app/validation.md'. |   | 10/07/2019 10:48:23
Error | User | include-not-found | aspnetcore/tutorials/razor-pages/validation.md | Cannot resolve '~/includes/currency.md' relative to 'tutorials/razor-pages/validation.md'.

**Comment by hishamco** [comment]
Oops, I thought it's used in the localization only. I will fix it ASAP

**Comment by Rick-Anderson** [comment]
Click on Details
image
to see the build report.

**Comment by hishamco** [comment]
I'm checking if I miss something else

**Comment by hishamco** [comment]
It passes

File: aspnetcore/fundamentals/localization.md [hunk 1]
```diff
@@ -270,6 +268,23 @@ The [Accept-Language header](https://www.w3.org/International/questions/qa-accep
...
 6. Tap the language, then tap **Move Up**.
 
+::: moniker range=">= aspnetcore-3.0"
+### The Content-Language HTTP header
+
+The [Content-Language](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Language) an entity header, is used to describe the language(s) intended for the audience, so that it allows a user to differentiate according to the users' own preferred language. Entity headers are used in both, HTTP requests and responses.
+
+In ASP.NET Core 3.0 the `Content-Language` header can be added optionally by setting the property `ApplyCurrentCultureToResponseHeaders`.
+
+This will allow the `RequestLocalizationMiddleware` to set the `Content-Language` header with the `CurrentUICulture`, instead to manage the response header `Content-Language` yourself.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/localization.md]: Adding the `Content-Language` header:

* Allows the `RequestLocalizationMiddleware` to set the `Content-Language` header with the `CurrentUICulture`.
* Eliminates the need to set the response header `Content-Language` explicitly.

File: aspnetcore/includes/localization/unsupported-culture-log-level.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+> [!NOTE]
+> Prior to ASP.NET Core 3.0 any web application will write one log of type `LogLevel.Warning` per request if the requested culture is unsupported, which is trouble and make the logs huge over the time. This behavior has been changed in ASP.NET 3.0 and the `RequestLocalizationMiddleware` will write a log of type `LogLevel.Debug` instead to reduce the amount of logs in production.
```
Comment (Rick-Anderson) [inline aspnetcore/includes/localization/unsupported-culture-log-level.md]: > Prior to ASP.NET Core 3.0 web apps writes one log of type `LogLevel.Warning` per request if the requested culture is unsupported. Logging one `LogLevel.Warning` per request is can make large log files with redundant information. This behavior has been changed in ASP.NET 3.0. The `RequestLocalizationMiddleware` writes a log of type `LogLevel.Debug`, which reduces the size of production logs.

**Comment by hishamco** [comment]
Sorry Rick, I enable "Allow edits from maintainers" option, I will look to your notes

File: aspnetcore/fundamentals/localization.md [hunk 3]
```diff
@@ -270,6 +268,23 @@ The [Accept-Language header](https://www.w3.org/International/questions/qa-accep
 
 6. Tap the language, then tap **Move Up**.
 
+::: moniker range=">= aspnetcore-3.0"
+### The Content-Language HTTP header
+
+The [Content-Language](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Language) an entity header, is used to describe the language(s) intended for the audience, so that it allows a user to differentiate according to the users' own preferred language. Entity headers are used in both, HTTP requests and responses.
```
[the remark this answers is not in the document]
Comment (hishamco) [reply aspnetcore/fundamentals/localization.md]: I enable it

File: aspnetcore/fundamentals/localization.md [hunk 4]
```diff
@@ -270,6 +268,23 @@ The [Accept-Language header](https://www.w3.org/International/questions/qa-accep
...
+
+In ASP.NET Core 3.0 the `Content-Language` header can be added optionally by setting the property `ApplyCurrentCultureToResponseHeaders`.
+
```
[the remark this answers is not in the document]
Comment (hishamco) [reply aspnetcore/fundamentals/localization.md]: Ya you have an option to add it by toggling the `ApplyCurrentCultureToResponseHeaders` property

**Comment by hishamco** [comment]
If you like to push your commits, feel free

File: aspnetcore/fundamentals/localization.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (hishamco) [reply aspnetcore/fundamentals/localization.md]: > I realize you got that from Monzilla, but long sentences don't MT (Machine translate)

That's why I refer it to Mozilla

File: aspnetcore/fundamentals/localization.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rick-Anderson) [reply aspnetcore/fundamentals/localization.md]: You have the option to set any property of any app. Let's drop the option talk.

**Comment by Rick-Anderson** [comment]
> If you like to push your commits, feel free, or I will do it in your behalf

I don't have permission. Just take my changes.

File: aspnetcore/fundamentals/localization.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (hishamco) [reply aspnetcore/fundamentals/localization.md]: What I'd like to refer here is, the `Content-Language` is optionally added to the response, by toggle the `ApplyCurrentCultureToResponseHeaders`

[the remark this answers is not in the document]
Comment (Rick-Anderson) [reply aspnetcore/fundamentals/localization.md]: OK. That's better wording.

`Content-Language` is optionally added to the response, by toggling the `ApplyCurrentCultureToResponseHeaders`

**Comment by hishamco** [comment]
Oops, it's OK now or I will take your changes

thanks

**Comment by Rick-Anderson** [comment]
Take my changes except add in the optional part.

**Comment by Rick-Anderson** [comment]
@ryanbrandenburg  see [Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/localization?view=aspnetcore-3.0&branch=pr-en-us-14874) 
and please review

**Comment by hishamco** [comment]
⌚️ 📆

File: aspnetcore/fundamentals/localization.md [hunk 5]
```diff
@@ -270,6 +268,31 @@ The [Accept-Language header](https://www.w3.org/International/questions/qa-accep
...
 
+::: moniker range=">= aspnetcore-3.0"
+### The Content-Language HTTP header
+
+The [Content-Language](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Language) entity header:
+
+ - Is used to describe the language(s) intended for the audience.
+ - Allows a user to differentiate according to the users' own preferred language.
+
+Entity headers are used in both, HTTP requests and responses.
```
Comment (ryanbrandenburg) [inline aspnetcore/fundamentals/localization.md]:
```suggestion
Entity headers are used in both HTTP requests and responses.
```

File: aspnetcore/fundamentals/localization.md [hunk 6]
```diff
@@ -336,7 +359,11 @@ Terms:
 * Parent culture: The neutral culture that contains a specific culture. (for example, "en" is the parent culture of "en-US" and "en-GB")
 * Locale: A locale is the same as a culture.
 
-[!INCLUDE[](~/includes/currency.md)]
+[!INCLUDE[](~/includes/localization/currency.md)]
```
Comment (ryanbrandenburg) [inline aspnetcore/fundamentals/localization.md]: Within this file change `This GitHub issue...` to `See this GitHub issue...`.

**Comment by hishamco** [comment]
:shipit:

**Pull request merged**